src/Appender/FileAppender.cs

```diff
@@ -876,6 +886,57 @@ public override void OnClose()
 		}
 #endif
 
+		/// <summary>
+		/// Method for secure open or create Mutex with Synchronize and Modify rights. 
+		/// Using this method allows different users run processes which can open or create a shared mutex without experiencing UnauthorizedAccessException.
+		/// </summary>
+		/// <param name="mutexId">The mutex Id. Here we use a mutex freindly name.</param>
+		/// <param name="errorHandler">Optional error handler, to log potentialy occuring errors.</param>
+		/// <returns></returns>
+		protected static Mutex SecureCreateMutex(string mutexId, IErrorHandler errorHandler)
+		{
+			// Using this mehotd to avoid the issue:
+			// Mutex ~ Access to the path is denied in log4net.Appender.RollingFileAppender.ActivateOptions()
+			// https://jira.apache.org/jira/browse/LOG4NET-587
+			// Solution implemented according to:
+			// UnauthorizedAccessException when trying to open a mutex
+			// https://stackoverflow.com/questions/19536697/unauthorizedaccessexception-when-trying-to-open-a-mutex
+			//NOTE: there are several other jira issues reporting this problem.
+			if (string.IsNullOrWhiteSpace(mutexId))
+			{
+				return null;
+			}
+
+			try
+			{
+				bool createdNew;
+				MutexSecurity mutexSecurity = new MutexSecurity();
+				mutexSecurity.AddAccessRule(
+					new MutexAccessRule(
+						new SecurityIdentifier(WellKnownSidType.WorldSid, null),
+						MutexRights.Synchronize | MutexRights.Modify,
+						AccessControlType.Allow
+						)
+					);
+
+				// attempt to create the mutex, with the desired DACL..
+				Mutex createdMutex = new Mutex(false, mutexId, out createdNew, mutexSecurity);
+				return createdMutex;
+			}
+			catch (WaitHandleCannotBeOpenedException ex)
+			{
+				// the mutex cannot be opened, probably because a Win32 object of a different type with the same name already exists.
+				errorHandler?.Error($"The mutex '{mutexId}' cannot be opened, probably because a Win32 object of a different type with the same name already exists.", ex);
```

Comment:
Is this a situation where the application is still in a safe state to continue working happily? What can be done to fix the situation when this happens? It would be great if the log message included one or more hints what a user of the logging framework could do to fix this situation.

Reply:
hm, this is a good question, to which I do not have an answer.
The original code did not catch the exception, thus letting the whole Appender to crash during Activate. (I think the exception was even thrown to the user code initializing log4net.) The Appender was logging nothing at all and causing all the issues like the LOG4NET-587.

If I "silently" allow to proceed (just with the invocation of the ErrorHandler), there will be another kind of issues, related with not locking. The usage of mutex itself is protected by querying of not null.

I improve the messages a bit. Give me feedback then, what do you think I shall do best.

Reply:
As far as I am concerned, applications can always be configured to log to different files and thus I would be happy with a simple file appender without locking or other fancy.